flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/KafkaSourceBuilder.java

```diff
@@ -481,5 +489,24 @@ private void sanityCheck() {
                 "No subscribe mode is specified, "
                         + "should be one of topics, topic pattern and partition set.");
         checkNotNull(deserializationSchema, "Deserialization schema is required but not provided.");
+        // Check consumer group ID
+        checkState(
+                props.containsKey(ConsumerConfig.GROUP_ID_CONFIG) || !offsetCommitEnabledManually(),
+                String.format(
+                        "Property %s is required when offset commit is enabled",
+                        ConsumerConfig.GROUP_ID_CONFIG));
+    }
```

Comment:
Ideally, we should probably also throw exception if the committedOffsets initializer is used when group id was not specified. It is a little tricky because `ReaderHandledOffsetsInitializerader` is package private. 

One possible solution is to have an `OffsetInitializerValidator` interface. If an `OffsetInitializer` needs to be validated, it can implement that interface and the `KafkaSourceBuilder` will then validate the properties against that `OffsetInitializer`. At the beginning this interface could be private. We may consider exposing it to end users later if that turns out to be necessary.

Reply:
Thanks for the suggestion @becketqin ! Implemented in 264a99c